librz/asm/arch/z80/z80asm.c

```diff
@@ -1055,105 +1051,105 @@ static int assemble(const char *str, unsigned char *_obuf) {
 				break;
 			}
 			switch (r) {
-			case ld_BC:
-			case ld_DE:
-				if (!rd_a (&ptr)) {
-					break;
-				}
-				wrtb (0x02 + 0x10 * (r == ld_DE));
-				break;
-			case ld_HL:
-				r = rd_ld_hl (&ptr);
-				wrtb (0x70 + --r);
-				break;
-			case ld_NN:
-				if (!(r = rd_ld_nn (&ptr))) {
-					break;
-				}
-				if (r == ld_nnA || r == ld_nnHL) {
-					wrtb (0x22 + 0x10 * (r == ld_nnA));
-					break;
-				}
-				wrtb (0xED);
-				wrtb (0x43 + 0x10 * --r);
-				break;
-			case ldA:
-				if (!(r = rd_lda (&ptr))) {
+				case ld_BC:
+				case ld_DE:
+					if (!rd_a (&ptr)) {
+						break;
+					}
+					wrtb (0x02 + 0x10 * (r == ld_DE));
 					break;
-				}
-				if (r == A_NN) {
-					wrtb (0x3A);
+				case ld_HL:
+					r = rd_ld_hl (&ptr);
+					wrtb (0x70 + --r);
 					break;
-				}
-				if (r == A_I || r == A_R) {
+				case ld_NN:
+					if (!(r = rd_ld_nn (&ptr))) {
+						break;
+					}
+					if (r == ld_nnA || r == ld_nnHL) {
+						wrtb (0x22 + 0x10 * (r == ld_nnA));
```

Comment:
please avoid doing multiplication with compare. split this or do `(r == ld_nnA ? 1 : 0)`